lib/PrintAsObjC/ModuleContentsWriter.cpp

```diff
@@ -56,7 +56,7 @@ class ReferencedTypeFinder : public TypeDeclFinder {
   Action visitTypeAliasType(TypeAliasType *aliasTy) override {
     if (aliasTy->getDecl()->hasClangNode() &&
         !aliasTy->getDecl()->isCompatibilityAlias()) {
-      assert(!aliasTy->getGenericSignature());
+      assert(!aliasTy->getDecl()->isGeneric());
```

Comment:
Since we can only get here if the typealias was originally imported from Clang, this looks sound to me—the parent generic context can't matter.